SubsoniciousKit/SubsoniciousKit/NowPlayingInfo/NowPlayingInfoManager.swift

```diff
@@ -38,51 +50,59 @@ public final class NowPlayingInfoManager {
-                                                  currentLanguageOptions: [],
-                                                  availableLanguageOptionGroups: [])
-
-        nowPlayingInfoCenter.setDynamicMetadata(metadata)
+        dynamicMetadata = NowPlayingMetadata.Dynamic(
+            rate: player.rate,
+            defaultRate: 1.0,
+            position: NSNumber(value: CMTimeGetSeconds(player.currentTime())),
+            currentLanguageOptions: [],
+            availableLanguageOptionGroups: [])
     }
-}
-
-extension MPNowPlayingInfoCenter {
-
-    func setStaticMetadata(_ metadata: NowPlayingMetadata.Static) {
-        var nowPlayingInfo = [String: Any]()
-
-        //        MPNowPlayingInfoPropertyCurrentPlaybackDate
-        //        MPNowPlayingInfoPropertyExternalContentIdentifier
-        //        MPNowPlayingInfoPropertyExternalUserProfileIdentifier
-        //        MPNowPlayingInfoPropertyPlaybackProgress
-        //        MPNowPlayingInfoPropertyServiceIdentifier
-        nowPlayingInfo[MPNowPlayingInfoCollectionIdentifier] = metadata.collectionIdentifier
-        nowPlayingInfo[MPNowPlayingInfoPropertyAssetURL] = metadata.assetURL
-        nowPlayingInfo[MPNowPlayingInfoPropertyMediaType] = metadata.mediaType.rawValue
-        nowPlayingInfo[MPNowPlayingInfoPropertyIsLiveStream] = metadata.isLiveStream
-        nowPlayingInfo[MPMediaItemPropertyTitle] = metadata.title
-        nowPlayingInfo[MPMediaItemPropertyArtist] = metadata.artist
-        nowPlayingInfo[MPMediaItemPropertyArtwork] = metadata.artwork
-        nowPlayingInfo[MPMediaItemPropertyAlbumArtist] = metadata.albumArtist
-        nowPlayingInfo[MPMediaItemPropertyAlbumTitle] = metadata.albumTitle
-        nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] = metadata.duration
-        nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackQueueCount] = metadata.queueCount
-        nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackQueueIndex] = metadata.queueIndex
-
-        self.nowPlayingInfo = nowPlayingInfo
-    }
-
-    func setDynamicMetadata(_ metadata: NowPlayingMetadata.Dynamic) {
-        var nowPlayingInfo = self.nowPlayingInfo ?? [String: Any]()
-
-        nowPlayingInfo[MPNowPlayingInfoPropertyElapsedPlaybackTime] = metadata.position
-        nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackRate] = metadata.rate
-        nowPlayingInfo[MPNowPlayingInfoPropertyDefaultPlaybackRate] = metadata.defaultRate
-        nowPlayingInfo[MPNowPlayingInfoPropertyCurrentLanguageOptions] = metadata.currentLanguageOptions
-        nowPlayingInfo[MPNowPlayingInfoPropertyAvailableLanguageOptions] = metadata.availableLanguageOptionGroups
 
-        self.nowPlayingInfo = nowPlayingInfo
+    func updateStaticMetadata() {
+        guard let currentItem = player.currentItem else { return }
+        let asset = currentItem.asset
+        let commonMetadata = asset.commonMetadata
+
+        let mediaType = MPNowPlayingInfoMediaType(
```

Comment:
I see a lot of `AVMetadataItem.metadataItems(from: commonMetadata...` maybe this could be mutualized (this should be according to the signature of the method with all you want fas property :)

Reply:
Good idea :)
Fixed in https://github.com/bilalb/Subsonicious/commit/5af0e03335014f487da313b23615cc2036075c23